### DISCUSSION TITLE
web oogle map on Supplierinfo model

### DISCUSSION THREAD

**Discussion by hetulhh** [gityopie/odoo-addons#214] [Q&A]
Hi Yopi, web_google_maps is an awesome module.
I need your help for an implementation. I am trying to implement the map on the Supplierinfo model in order to search and get on the map the products and its suppliers. The map loads ok, but when I filter for the supplier the sidebar gives me the correct information but the maps pinpoints all the suppliers locations present on the model Supplierinfo: it seems that its is not clearing the markers. Same happens when I try to filter for the products, the siderbar gives the correct information but the maps loads all the suppliers locations. Sending a example:

`from odoo import api, fields, models

class SupplierInfo(models.Model):
    _inherit = 'product.supplierinfo'

    #par_id = fields.Integer('Partner Parent ID',compute='_get_partner')

    def _get_partner(self):
        domain = []
        flag = self.env['res.users'].has_group('hh_multivendor.group_hh_multivendor_vendor')
        if flag:
            domain = [('id', '=', self.env.user.partner_id.parent_id.id)]
        return domain

    @api.depends('min_qty','partner_name','price')
    def _compute_subtitle(self):
        for record in self:
            record.subtitle = "Supplier: %s, Qt: %s, Price: %s" % (record.partner_name,str(record.min_qty),str(record.price))

    name = fields.Many2one(domain=_get_partner)
    id_partner =  fields.Integer(related='name.id')
    partner_name = fields.Char(related='name.name')
    partner_latitude = fields.Float(string='Supplier Latitude',related='name.partner_latitude',store=False, readonly=False, digits=(16, 5))
    partner_longitude = fields.Float(string='Supplier Longitude',related='name.partner_longitude',store=False, readonly=False, digits=(16, 5))
    color = fields.Char(
        string='Color', default='red', required=True)
    display_name =fields.Char(related='product_tmpl_id.name')
    title = fields.Many2one(related='name.title')
    email = fields.Char(related='name.email')
    parent_id = fields.Many2one(related='name.parent_id')
    is_company = fields.Boolean(related='name.is_company')
    function = fields.Char(related='name.function')
    phone = fields.Char(related='name.phone')
    street = fields.Char(related='name.street')
    street2 = fields.Char(related='name.street2')
    zip = fields.Char(related='name.street2')
    city = fields.Char(related='name.zip')
    country_id = fields.Many2one(related='name.country_id')
    mobile =  fields.Char(related='name.mobile')
    state_id = fields.Many2one(related='name.state_id')
    category_id = fields.Many2many(related='name.category_id')
    type = fields.Selection(related='name.type')
    marker_color = fields.Char(
        string='Marker Color', default='red', required=True)
    company_type = fields.Selection(related='name.company_type')
    image_128 = fields.Image(related='product_tmpl_id.image_128')
    seller_ids = fields.One2many(related='product_tmpl_id.seller_ids')
    subtitle = fields.Char(compute="_compute_subtitle")`

![Product Search](https://github.com/gityopie/odoo-addons/assets/395674/041650ae-1848-46c7-942f-c0406795f41e)
![Vendor Search](https://github.com/gityopie/odoo-addons/assets/395674/8c034f55-ea68-4f03-9964-6113259a8706)

Can you help?

Thanks,
Hetul

**Comment by gityopie** [comment]
Hi @hetulhh 

Did you disabled marker cluster on your google_map view?

Would be great if you can share your custom module so that I can check it

Thanks!

**Comment by hetulhh** [comment]
Hi Yopi,

I didn´t disable the marker cluster.
Please check attached the custom module.

Thank you for your help,

Hetul

Yopi Angi ***@***.***> escreveu no dia terça, 6/06/2023 à(s)
18:17:

**Reply by gityopie** [reply]
@hetulhh  I don't see any attachment

**Reply by hetulhh** [reply]
Please check the attached file.
[hh_multivendor.zip](https://github.com/gityopie/odoo-addons/files/11674169/hh_multivendor.zip)

**Comment by gityopie** [comment]
@hetulhh I've a pushed a fixes, please take the latest version and update web_google_maps on your instance.
Thank you for your feedback

**Reply by hetulhh** [reply]
@gityopie It is now working fine. Thank you for your help.
